### PR TITLE
Create ci.yaml

### PR DESCRIPTION
**Pull request by rahulrane50**
Creating ci.yaml file to run CI for ZK ACL dev branch

 
    
### Description
Adding ci.yaml file from master branch to this dev branch so that every PR would go through CI tests.

### Tests
Tried dispatching manual workflows on this branch.

The following is the result of the "mvn test" command on the appropriate module:
(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)
My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:
(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)
NA

### Documentation (Optional)
In case of new functionality, my PR adds documentation in the following wiki page:
(Link the GitHub wiki you added)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          - name: 'full-build-jdk11'
+            jdk: 11
+            args: '-Pfull-build apache-rat:check verify -DskipTests spotbugs:check checkstyle:check'
+          - name: 'full-build-java-tests'
+            jdk: 11
+            args: '-Pfull-build verify -Dsurefire-forkcount=1 -DskipCppUnit -Dsurefire.rerunFailingTestsCount=5'
+          - name: 'full-build-cppunit-tests'
+            jdk: 11
+            args: '-Pfull-build verify -Dtest=_ -DfailIfNoTests=false'
+      fail-fast: false
```
Comment (micahstubbs) [inline .github/workflows/ci.yaml:51]: Curious, why don't we want to fail fast?

**Review by micahstubbs — COMMENTED** [review]
One question about `fail_fast`, otherwise look great.  Thanks for the contribution!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rahulrane50) [reply .github/workflows/ci.yaml:51]: From my understanding, zk tests are bit flaky and they mostly pass when retried. Hence we don't want to fail on first failure but try everything then re-try failed ones and then see the status for run.

**Review by mgao0 — COMMENTED** [review]
In the CI results run for this PR, there's no detailed failed tests log. Could you please check why?

**Review by mgao0 — APPROVED** [review]
LGTM

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (micahstubbs) [reply .github/workflows/ci.yaml:51]: Ah, I see. Thanks for explaining.

**Pull request merged**